citest/service_testing/http_predicate.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""HTTP Response Predicates."""
+
+
+import re
+import citest.json_predicate as jp
+
+from . import HttpResponseType
+
+
+class HttpResponsePredicateResult(jp.PredicateResult):
+  """Result from an HttpResponsePredicate."""
+
+  @property
+  def value(self):
+    """The value the predicate was applied to."""
+    return self.__value
+
+  def __init__(self, valid, value, **kwargs):
+    super(HttpResponsePredicateResult, self).__init__(valid, **kwargs)
+    self.__value = value
+
+  def __eq__(self, result):
+    return (super(HttpResponsePredicateResult, self).__eq__(result)
+            and self.__value == result.value)
+
+  def export_to_json_snapshot(self, snapshot, entity):
+    """Implements JsonSnapshotableEntity interface."""
+    builder = snapshot.edge_builder
+    builder.make_output(entity, 'HTTP Code', self.__value.http_code)
```

Comment:
Is there a request path we can add to this?

Reply:
This typically represents the value that was compared with the predicate, not its pedigree.
If the source were of interest, then it would probably also be elsewhere in the snapshot context.
For example if this is an observed value, the observation request would be in the snapshot.
The path would be there.